### ISSUE TITLE
[Bug] Login Hint and Prompt are ignored

### ISSUE THREAD

**Issue by boboldi** [AzureAD/microsoft-authentication-library-for-dotnet#1837]
**Which Version of MSAL are you using ?**
v4.11.0

**Platform**
Xamarin for Android

**What authentication flow has the issue?**
* Desktop / Mobile
    * [x] Interactive

**Is this a new or existing app?**
The app is in production, and I have upgraded to a new version of MSAL

**Repro**

```public static void ConfigureApi()
{
	var aadInstance = CurrentEnvironment.AadInstance;
	var tenantId = CurrentEnvironment.TenantId;
	var authority = new Uri(string.Format(CultureInfo.InvariantCulture, aadInstance, tenantId));
	var signatureHash = CurrentEnvironment.SignatureHash;
	var versionString = (DeviceInfo.VersionString.IndexOf('.') > 0) ? DeviceInfo.VersionString.Substring(0, DeviceInfo.VersionString.IndexOf('.')) : DeviceInfo.VersionString;
	var versionNumber = int.Parse(versionString, CultureInfo.InvariantCulture);
	if (Xamarin.Forms.Device.RuntimePlatform == Xamarin.Forms.Device.iOS && versionNumber >= 13)
	{
		PCA = PublicClientApplicationBuilder
			.Create(ClientId)
			.WithBroker(!DevEnvironment)
			.WithRedirectUri($"<value here>")
			.WithParentActivityOrWindow(() => ParentWindow)
			.WithAuthority(authority)
			.WithIosKeychainSecurityGroup(Consts.SecureStorageGroup)
			.Build();
	}
	else if (Xamarin.Forms.Device.RuntimePlatform == Xamarin.Forms.Device.iOS)
	{
		PCA = PublicClientApplicationBuilder
			.Create(ClientId)
			.WithBroker(false)
			.WithRedirectUri($"<value here>")
			.WithParentActivityOrWindow(() => ParentWindow)
			.WithAuthority(authority)
			.WithIosKeychainSecurityGroup(Consts.SecureStorageGroup)
			.Build();
	}
	else
	{
		PCA = PublicClientApplicationBuilder
			.Create(ClientId)
			.WithBroker(true)
			.WithRedirectUri($"<value here>")
			.WithParentActivityOrWindow(() => ParentWindow)
			.WithAuthority(authority)
			.Build();
	}
}
```

```public async Task<AuthData> Login(bool forceLogin, CancellationToken ct)
{
	bool waiting = false;

	try
	{
		await _accessTokenLock.WaitAsync(ct);
		waiting = true;

		if ((_authData != null) && !forceLogin)
		{
			return _authData;
		}

		var scopes = SCOPES.Select(s => $"api://{App.ClientId}/{s}");

		var authResult = await GetTokenAsync(scopes, null, forceLogin).ConfigureAwait(false);
		if (authResult != null)
		{
			_authData = await SaveAuthResult(authResult);
			return _authData;
		}
	}
	catch (MsalServiceException msex)
	{
		// Only handle error when it is not cancel from user
		if (msex.ErrorCode != MsalError.AuthenticationCanceledError)
		{
			_logger.LogError(msex);

			var message = $"{Strings.AuthError_General}\n\nMsalServiceException Details:\nErrorCode={msex.ErrorCode}\nMessage={msex.Message}";

			// Special handling for network not available
			if (msex.ErrorCode == "device_network_not_available")
			{
				message = Strings.AuthError_NoNetwork;
			}

			throw new AuthException(message);
		}
	}
	catch (MsalClientException mcex)
	{
		// Only handle error when it is not cancel from user
		if (mcex.ErrorCode != MsalError.AuthenticationCanceledError)
		{
			_logger.LogError(mcex);

			var message = $"{Strings.AuthError_General}\n\nMsalClientException Details:\nErrorCode={mcex.ErrorCode}\nMessage={mcex.Message}";

			// Special handling for network not available
			if (mcex.ErrorCode == "device_network_not_available")
			{
				message = Strings.AuthError_NoNetwork;
			}

			throw new AuthException(message);
		}
	}
	catch (Exception ex)
	{
		_logger.LogError(ex);
		var message = $"{Strings.AuthError_General}\n\nDetails:\nMessage={ex.Message}";
		throw new AuthException(message);
	}
	finally
	{
		if (waiting)
		{
			_accessTokenLock.Release();
		}
	}

	return null;
}

private async Task<AuthenticationResult> GetTokenAsync(IEnumerable<string> scopes, string claims = null, bool forceLogin = false, IAccount account = null)
{
	var correlationId = Guid.Parse(CORRELATION_ID);
	var builder = App.PCA.AcquireTokenInteractive(scopes).WithCorrelationId(correlationId);

	if (DeviceInfo.Platform == DevicePlatform.iOS && DeviceInfo.Version.Major >= 13)
	{
		builder = builder.WithUseEmbeddedWebView(true);
	}
	else
	{
		builder = builder.WithParentActivityOrWindow(App.ParentWindow);
	}

	if (account != null)
	{
		builder = builder.WithAccount(account);
	}

	if (!string.IsNullOrEmpty(claims))
	{
		builder = builder.WithClaims(claims);
	}

	if (forceLogin)
	{
		if (App.UserInfo != null && !string.IsNullOrEmpty(App.UserInfo.EmailAddress))
		{
			builder = builder.WithLoginHint(App.UserInfo.EmailAddress);
		}

		builder = builder.WithPrompt(Prompt.ForceLogin);
	}

	var authResult = await builder.ExecuteAsync().ConfigureAwait(false);
	if (authResult != null)
	{
		App.UserInfo = new Users.UserInfo
		{
			Id = authResult.UniqueId,
			EmailAddress = authResult.Account.Username
		};
	}

	return authResult;
}
```

**Expected behavior**
Passing a login hint and a prompt to force login to the broker should cause an interactive prompt and ask the user for credentials.

**Actual behavior**
Users are not getting prompted to provide their credentials 

**Possible Solution**
Make sure that MSAL is passing both login hint and prompt to the broker

**Comment by jennyf19** [comment]
@boboldi thank you .

**Comment by trwalke** [comment]
@boboldi this was a known bug that was recently resolved in the latest MSAL 4.14 release
https://www.nuget.org/packages/Microsoft.Identity.Client/4.14.0

can you try updating and signing in?
